deduce.go

```diff
@@ -519,76 +524,166 @@ func (m vcsExtensionDeducer) deduceSource(path string, u *url.URL) (maybeSource,
 	}
 }
 
-type stringFuture func() (string, error)
-type sourceFuture func() (source, string, error)
-type partialSourceFuture func(string, ProjectAnalyzer) sourceFuture
+type deductionCoordinator struct {
+	callMgr  *callManager
+	rootxt   *radix.Tree
+	deducext *deducerTrie
+	action   chan func()
+}
+
+func newDeductionCoordinator(cm *callManager) *deductionCoordinator {
+	dc := &deductionCoordinator{
+		callMgr:  cm,
+		rootxt:   radix.New(),
+		deducext: pathDeducerTrie(),
+		action:   make(chan func()),
```

Comment:
Yeah, I think this is just more of the actor mentality hanging around. Seemed like such a good idea at first, but it just doesn't seem that helpful here (non-exclusive or I'm doing it wrong 😄 ).

I'll probably change this to a mutex, as well.